File: Application/social_sense/server/models/UserModel.js
const mongoose = require("mongoose");

const userSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  password: {
    required: true,
    type: String,
  },
  email: {
    type: String,
    required: true,
    unique: true,
  },
  profilepic: {
    type: String,
  },
  bio: {
    type: String,
  },
  phone: {
    type: String,
  },
  dob: {
    type: String,
  },
  posts: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Post",
    },
  ],
});

module.exports = User = mongoose.model("User", userSchema);

File: Application/social_sense/lib/models/user_model.dart
class UserModel {
  final String id;
  final String? name;
  final String? email;
  final String? dob;
  final String? bio;
  final String? phone;

  UserModel({
    required this.id,
    this.name,
    this.email,
    this.dob,
    this.bio,
    this.phone,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['_id'],
      name: json['name'],
      email: json['email'],
      dob: json['dob'],
      bio: json['bio'],
      phone: json['phone'],
    );
  }

  @override
  String toString() {
    return 'User{id: $id, name: $name, email: $email, dob: $dob, bio: $bio, phone: $phone}';
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'email': email,
      'dob': dob,
      'bio': bio,
      'phone': phone,
    };
  }
}
